packages/aws-cdk-lib/aws-apigateway/lib/lambda-api.ts

```diff
@@ -68,6 +69,22 @@ export class LambdaRestApi extends RestApi {
       this.root.addMethod = addMethodThrows;
       this.root.addProxy = addProxyThrows;
     }
+
+    this.node.addValidation({
+      validate() {
+        if (props.deployOptions?.variables) {
+          for (let key in props.deployOptions.variables) {
+            // Checks that variable Stage values match regex
+            const regexp = /[A-Za-z0-9-._~:/?#&amp;=,]+/;
```

Comment:
Is this regex correct? The docs have this one `[A-Za-z0-9-._~:/?#&=,]+`

Reply:
Typo on my end - thanks for catching this!